redisinsight/api/src/modules/database/repositories/local.database.repository.ts

```diff
@@ -90,6 +92,17 @@ export class LocalDatabaseRepository extends DatabaseRepository {
    * @param database
    */
   public async create(database: Database): Promise<Database> {
+    // Do not create a connection if it triggered from cloud and have the same fields
+    if (database.cloudDetails) {
+      const existingEntity = await this.findEntity(
+        database,
+        ['host', 'port', 'username', 'password', 'tls'],
+      );
+      if (existingEntity) {
```

Comment:
based on "findEntity" function I assume that you forgot encrypt fields before comparing values. Here you are receiving  "Database" object with plain values when "Database Entity" stores encrypted values.

Reply:
added encryption